packages/middleware/src/apiClientFactory/index.ts

```diff
@@ -82,13 +74,30 @@ const apiClientFactory = <
         )
         .reduce(async (configSoFar, extension) => {
           const resolvedConfig = await configSoFar;
+          const loggerWithMetadata = injectMetadata(logger, (metadata) => ({
+            ...metadata,
+            scope: {
+              ...metadata?.scope,
+              extensionName: extension.name,
+              hookName: "beforeCreate",
+              type: "requestHook",
+            },
+          }));
```

Comment:
This logic is the same for all hooks but the hook name changes, can we extract it into a function to avoid c/p and simplify the code?